aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1359,127 +1356,110 @@ Tensor orgqr(const Tensor& input, const Tensor& tau) {
   return result;
 }
 
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ syevd ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ linalg_eigh ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-// This function computes eigenvalues 'w' and eigenvectors 'v' of the input that is stored initially in 'v'
-// The computation is done in-place: 'v' stores the input and will be overwritten, 'w' should be an allocated empty array
-// compute_v controls whether eigenvectors should be computed
-// uplo_str controls the portion of input matrix to consider in computations, allowed values are "u", "U", "l", "L"
-// infos is used to store information for possible checks for error
-// This function doesn't do any error checks and it's assumed that every argument is valid
-template <typename scalar_t>
-static void apply_syevd(Tensor& w, Tensor& v, bool compute_v, const std::string& uplo_str, std::vector<int64_t>& infos) {
-#ifndef USE_LAPACK
-  AT_ERROR("syevd: LAPACK library not found in compilation");
-#else
-  using value_t = typename c10::scalar_value_type<scalar_t>::type;
+DEFINE_DISPATCH(linalg_eigh_stub);
 
-  auto v_data = v.data_ptr<scalar_t>();
-  auto w_data = w.data_ptr<value_t>();
-  auto v_matrix_stride = matrixStride(v);
-  auto w_stride = w.size(-1);
-  auto batch_size = batchCount(v);
-  auto n = v.size(-1);
-  auto lda = std::max(int64_t{1}, n);
+/*
+  Computes eigenvalues and eigenvectors of the tensor 'input'.
 
-  // NumPy allows lowercase input for UPLO argument
-  // It is assumed that uplo_str is either "U" or "L"
-  char uplo = std::toupper(uplo_str[0]);
-  char jobz = compute_v ? 'V' : 'N';
+  Args:
+  * 'input' - input Tensor for eigendecomposition
+  * 'values' - Tensor to store computed eigenvalues
+  * 'vectors' - Tensor to store computed eigenvectors
+  * 'infos' - Tensor to store LAPACK/MAGMA/cuSOLVER error codes
+  * 'compute_eigenvectors' - controls whether eigenvectors should be computed
+  * 'uplo_str' - controls the portion of input matrix to consider in computations, allowed values are "u", "U", "l", "L"
```

Comment:
This comment is great, but add a sentence for what the different uplo_str values actually mean.